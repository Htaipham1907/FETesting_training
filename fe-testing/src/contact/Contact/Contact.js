import React, { Component } from "react";
import ContactForm from './ContactForm/ContactForm';
import Header from '../../Header/Header'
import Footer from '../../Footer/Footer'
import './Contact.scss';

class Contact extends Component{
    render(){
        return(
            <React.Fragment>
                <Header></Header>
                <section>
                    <ContactForm></ContactForm>
                </section>
                <Footer></Footer>
            </React.Fragment>
        )
    }
}

export default Contact;