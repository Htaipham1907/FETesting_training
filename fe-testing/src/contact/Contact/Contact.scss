section{
    background: url(../../resource/images/body-bg.jpeg);
    height:130vh;
    background-size: cover;
    display:flex;
    align-items: center;
    justify-content: center;
}