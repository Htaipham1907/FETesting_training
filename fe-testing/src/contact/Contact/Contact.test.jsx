import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor,mount, getByPlaceholderText, queryByRole, queryByTestId, getByTestId, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Contact from './Contact';


// //snapshot
// test('match snapshot', () => {
//   const tree = renderer.create(<Contact/>).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// //simulate
// test('display err text when user did not type in input field', async () => {
//     const { getByPlaceholderText, getByText } = render(<Contact />);
//     const input = getByPlaceholderText('Enter your issue')
//     fireEvent.blur(input);
//     let error;
//     await waitFor(() => {
//       error = getByText('Required field');
//     });
//     expect(error).toBeTruthy();
//   });


// test('display err text when user did not type in message field', async () => {
//     const { getByPlaceholderText, getByText } = render(<Contact />);
//     const message = getByPlaceholderText('Enter your message')
//     fireEvent.blur(message);
//     let error;
//     await waitFor(() => {
//       error = getByText('Required field');
//     });
//     expect(error).toBeTruthy();
//   });


// // expected output 
// test('input field recieve correct value from user', async () => {
//     const { getByPlaceholderText } = render(<Contact />);
//     const input = getByPlaceholderText('Enter your issue')
//     await userEvent.type(input, 'Tai Pham', { delay: 1 });
//     expect(input.value).toBe('Tai Pham');
//   });


it("Form can be submited", () => {
    const mockSubmit = jest.fn();
    const {getByTestId } = render(<Contact onSubmit={mockSubmit}/>);
    act(() => {
        fireEvent.change(getByTestId("input-test"), { target: { value: 'Tai Pham' } });
        fireEvent.submit(getByTestId("my-form"));
        expect(mockSubmit).toHaveBeenCalled(); 
        expect(mockSubmit.mock.calls).toEqual([[{name: 'Tai Pham'}]]); 
    });
});