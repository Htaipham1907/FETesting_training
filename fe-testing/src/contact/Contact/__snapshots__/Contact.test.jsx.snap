// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
Array [
  <div
    className="myHeader"
  >
    <img
      alt=" "
      id="img-logo"
      src="logo.webp"
    />
    <i
      className="fas fa-bars"
      id="menu"
    />
  </div>,
  <section>
    <form
      className="form-container"
      data-testid="my-form"
      onSubmit={[Function]}
    >
      <div
        className="col1"
      >
        <div
          className="namefield"
        >
          <label>
            Your name
          </label>
          <input
            data-testid="input-test"
            name="yourname"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter your issue"
            type="text"
            value=""
          />
        </div>
        <div
          className="selectfield"
        >
          <label>
            Select
          </label>
          <select
            name="typedrive"
            onBlur={[Function]}
            onChange={[Function]}
            value="Rider"
          >
            <option
              value="Rider"
            >
              Rider
            </option>
            <option
              value="Driver"
            >
              Driver
            </option>
          </select>
        </div>
      </div>
      <div
        className="col2"
      >
        <div
          className="messagefield"
        >
          <label>
            Message
          </label>
          <textarea
            name="message"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter your message"
            value=""
          />
        </div>
      </div>
      <button
        className="formButton"
      >
        <i
          className="fa fa-arrow-right"
        />
      </button>
    </form>
  </section>,
  <div
    className="foot-container"
  >
    <hr />
    <div
      className="myFooter"
    >
      <div
        className="content"
      >
        <p>
          Â© 2019 Technologies Inc. All Rights Reserved
          <br />
           Term of use   |   Local Notices   |   Privacy & Security
        </p>
      </div>
      <div
        className="social-icon"
      >
        <a
          className="fa fa-facebook"
          href=" "
        />
        <a
          className="fa fa-linkedin"
          href=" "
        />
        <a
          className="fa fa-youtube"
          href=" "
        />
      </div>
    </div>
  </div>,
]
`;
